changelog/fragments/k8s-1.24-bump.yaml

```
@@ -0,0 +1,110 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+
+# TODO(everettraven): update changelog
+entries:
+  - description: >
+      For Go (go/v3) and Helm Hybrid (hybrid) Operators: Upgrade Kubernetes 1.23 to 1.24
+
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "change"
+
+    # Is this a breaking change?
+    breaking: false
+
+    # Migration can be defined to automatically add a section to
+    # the migration guide. This is required for breaking changes.
+    migration:
+      header: Go (go/v3) and Hybrid Helm (hybrid) Operators - Upgrading to Kubernetes 1.24
+      body: |
+        To update your project to support Kubernetes 1.24:
+        
+        1. In `go.mod`:
+
+        - Replace `k8s.io/api v0.23.5` with `k8s.io/api v0.24.0`
+        - Replace `k8s.io/apimachinery v0.23.5` with  `k8s.io/apimachinery v0.24.0`
+        - Replace `k8s.io/client-go v0.23.5` with `k8s.io/client-go v0.24.0`
+        - Replace `sigs.k8s.io/controller-runtime v0.11.2` with `sigs.k8s.io/controller-runtime v0.12.1`
+        - Hybrid Helm (hybrid) only: Replace `github.com/operator-framework/helm-operator-plugins v0.0.11` with `github.com/operator-framework/helm-operator-plugins v0.0.12-0.20220608155702-d3967d2ae2ac`
+
+        2. Run `go mod tidy` to ensure all dependencies are upgraded
+        
+        3. In `Makefile`:
+        - Replace `CONTROLLER_TOOLS_VERSION ?= 0.8.0` with `CONTROLLER_TOOLS_VERSION ?= 0.9.0`
+        - Replace `ENVTEST_K8S_VERSION = 1.23` with `ENVTEST_K8S_VERSION = 1.24`
+
+        4. Run `make` to run all make targets
+
+  - description: >
+      For Go (go/v3) and Hybrid Helm (hybrid) Operators: Support for Go 1.18
+      
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "change"
+
+    # Is this a breaking change?
+    breaking: false
+
+    # Migration can be defined to automatically add a section to
+    # the migration guide. This is required for breaking changes.
+    migration:
+      header: Go (go/v3) and Hybrid Helm (hybrid) Operator - Update to Go 1.18
```

```suggestion
      header: For Golang (go/v3) and Helm Hybrid (hybrid.helm/v1-alpha) based-language operators - Add support for Go 1.18
```

we can use the same text. 1 is used for the release notes and another for the migration guide

---

commit https://github.com/operator-framework/operator-sdk/pull/5843/commits/d7cde371a82bf068ad88d9f34bf55e216d97b39e